elasticsearch/file/src/main/java/nl/knaw/huc/exception/TextRepoRequestExceptionMapper.java

```diff
@@ -0,0 +1,21 @@
+package nl.knaw.huc.exception;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+public class TextRepoRequestExceptionMapper implements ExceptionMapper<TextRepoRequestException> {
+
+  private static final Logger log = LoggerFactory.getLogger(TextRepoRequestExceptionMapper.class);
+
+  @Override
+  public Response toResponse(TextRepoRequestException ex) {
+    log.error(ex.getMessage(), ex);
+    return Response
+        .status(503)
```

Comment:
"The HyperText Transfer Protocol (HTTP) 503 Service Unavailable server error response code indicates that the server is not ready to handle the request."

Gegeven dat sommige endpoints een IllegalArgumentException gooien bij een verkeerde input parameter (via REST call) zou dat eerder een 400 Bad Request zijn?

Misschien is er ook een use case voor deze 503, maar dan moeten we duidelijk onderscheid maken tussen welke IllegalArgument nu een 400 en welke een 503 is. Het voelt voor mij alsof 503 een tijdelijke resource-exhaustion of zo is ("het bakje is vol, we kunnen nu even geen nieuwe knikkers aannemen")

Reply:
TextRepoRequestException wordt gegooid als de text repo een resultaat terug geeft dat niet verwacht wordt: in dat geval zal de TextRepo eruit liggen, of een foutmelding gegooid hebben, waarbij 400 me het meest logische leek